### ISSUE TITLE
text input waiting

### ISSUE THREAD

**Issue by markmbaum** [marimo-team/marimo#2218]
### Description

I frequently use the text and text_area inputs to define file names or paths, from which some data are loaded. Reactive cell execution is triggered every time I update these inputs, even a single character. This repeatedly triggers downloads and other overhead and slows down the whole process considerably.

### Suggested solution

I suggest implementing an optional delay period after which text inputs are considered finished and downstream cells are executed. For example, a text_area box can optionally be considered complete two seconds after the last update to its contents, after which the rest of the downstream cells are run.

**Comment by mscolnick** [comment]
hi @markmbaum thanks for the suggestion - this is a great idea. I was just talking with a contributor about this, we were thinking of adding "debounce=True" which will wait until the input is blurred (loses focus) or Enter is hit (Cmd+Enter for text area)

**Comment by markmbaum** [comment]
That would definitely work but I think it would be even more seamless to just add a slight delay for these situations. Then the user doesn't have to take any extra steps and remember to take any extra steps.

**Comment by mscolnick** [comment]
We can make debounce `True | int` to support a delay

**Comment by markmbaum** [comment]
That sounds like it would also work well

**Comment by wasimsandhu** [comment]
Picking this up today 🙂

**Comment by wasimsandhu** [comment]
Closed #2218 as completed.

**Comment by mscolnick** [comment]
Thanks @wasimsandhu 

@markmbaum, you can now debounce inputs:

```python
mo.ui.text(debounce=True)
# or
mo.ui.text(debounce=500) # ms

mo.ui.text_area(debounce=True)
# or
mo.ui.text_area(debounce=500) # ms
```

p.s. we may consider making this the default later
